cmd/server/main.go

```diff
@@ -1,7 +1,49 @@
 package main
 
-import "fmt"
+import (
+	"html/template"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+type User struct {
+	Email    string
+	Username string
+	Password string
+}
+
+func RegistrationFormHandler(w http.ResponseWriter, r *http.Request) {
+	tmpl, _ := template.ParseFiles("views/registration.html")
+	err := tmpl.Execute(w, nil)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+func RenderingRegistrationDetails(w http.ResponseWriter, r *http.Request) {
+	registrationDetails := User{
+		Email:    r.FormValue("email"),
+		Username: r.FormValue("username"),
+		Password: r.FormValue("password"),
+	}
+
+	_ = registrationDetails
+
+	tmpl, _ := template.ParseFiles("views/registration.html")
+	err := tmpl.Execute(w, struct{ Success bool }{true})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
 
 func main() {
-	fmt.Println("Hello, Identity Provider")
+	r := mux.NewRouter()
+	r.HandleFunc("/registration", RegistrationFormHandler).Methods("GET")
+	r.HandleFunc("/registration", RenderingRegistrationDetails).Methods("POST")
+	http.Handle("/", r)
```

Comment:
You don't need this `http.Handle` line, because it will register the router on the default mux in the `http` package. You are already passing it to the `http.ListenAndServe` function.